In [1]:
%pip install python-dotenv requests


[notice] A new release of pip is available: 26.0.1 -> 26.1.1
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


https://claude.ai/share/e972af20-6715-4a09-a266-4953dce2a625
사용한 llm링크

In [4]:
import os
from dotenv import load_dotenv

load_dotenv()
KEY: str = os.environ["KOREAN_DICT_KEY"]
print(KEY[:4]+'***')



4333***


In [26]:
#1번
import os, requests
from dotenv import load_dotenv

load_dotenv()
KEY: str = os.environ["KOREAN_DICT_KEY"]
def search_word(q: str, num: int = 10, start: int = 1) -> dict:
    params = {
        "key" : KEY, 
        "q" : q,
        "req_type" : "json",
        "num" : num,
        "start" : start,
        "type1": "word",
    }

    r = requests.get("https://opendict.korean.go.kr/api/search", params=params, timeout = 10)
    r.raise_for_status()
    data = r.json()
    return data
    


In [28]:
import json
data = search_word("김치")
print(json.dumps(data, ensure_ascii=False, indent=2)[:400])

{
  "channel": {
    "total": 328,
    "num": 10,
    "title": "우리말샘 개발 지원(Open API) - 사전 어휘 검색",
    "start": 1,
    "description": "우리말샘 개발 지원(Open API) - 사전 어휘 검색 결과",
    "link": "https://opendict.korean.go.kr",
    "item": [
      {
        "word": "김치",
        "sense": [
          {
            "syntacticArgument": "",
            "syntacticAnnotation": "",
            "cat": "",
          


In [33]:
items = data["channel"]["item"]
print(f"총{data['channel']['total']}건, 이 페이지{len(items)}건")

for it in items[:5]:
    word = it["word"]
    pos = it["sense"][0].get("pos", "품사 없음")
    sense = it["sense"][0] ["definition"]
    print(f"{word}({pos}): {sense[:40]}")

총227건, 이 페이지10건
국수(명사): 밀가루·메밀가루·감자 가루 따위를 반죽한 다음, 반죽을 손이나 기계 따위
국수(명사): 장기, 바둑 따위에서 그 실력이 한 나라에서 으뜸가는 사람.
국수(명사): 한 나라나 민족이 지닌 고유한 정신적·물질적인 장점.
국수(명사): 나라의 원수.
국수(명사): 두 손을 오목하게 만들어 물을 뜸. 또는 그렇게 뜬 물.


#1 실행결과
{
  "channel": {
    "total": 328,
    "num": 10,
    "title": "우리말샘 개발 지원(Open API) - 사전 어휘 검색",
    "start": 1,
    "description": "우리말샘 개발 지원(Open API) - 사전 어휘 검색 결과",
    "link": "https://opendict.korean.go.kr",
    "item": [
      {
        "word": "김치",
        "sense": [
          {
            "syntacticArgument": "",
            "syntacticAnnotation": "",
            "cat": "",
총328건, 이 페이지10건
김치(명사): 소금에 절인 배추나 무 따위를 고춧가루, 파, 마늘 따위의 양념에 버무린
김-치(명사): 고려 말기·조선 초기의 문신(?~?). 자는 기보(基甫). 김해 부사를 
김-치(명사): 조선 중기의 문신(1577~1625). 자는 사정(士精). 호는 남봉(南
김치 공장(): 김치를 만드는 공장.
김치 보릿고개(): 김장철인 가을·겨울과 달리 상대적으로 김치가 부족한 봄여름을 비유적으로 

코드 설명:
먼저 load_dotenv() 함수를 이용하여 api키를 읽는다.
requests를 이용하여 링크에 접근한다. 이떄 params를 이용하여 url의 쿼리 스트링을 생성한다.
items에 데이터를 저장하고 word에 단어 이름, pos에 품사 sense에 첫번째 뜻을 저장합니다. 

print(json.dumps(data, ensure_ascii=False, indent=2)[:400])코드에서 ensure_ascii=True를 할 경우 한글이 유니코드 이스케이프 시퀸스로 출력되어 사람이 읽기가 어려워집니다.

In [36]:
#2
import time

words : list[str] = ["김치", "라면", "만두", "김밥", "국수", "떡볶이", "불고기", "비빔밥"]


for w in words:
    data = search_word(w)
    print(f"{w}:{data['channel']['total']}건")
    time.sleep(0.3)

김치:328건
라면:86건
만두:89건
김밥:39건
국수:227건
떡볶이:24건
불고기:38건
비빔밥:38건


In [34]:
import collections

lst = []

for w in words:
    time.sleep(0.3)
    data = search_word(w)
    items = data["channel"]["item"]
    
    for item in items:
        pos = item["sense"][0].get("pos") or "(미상)"
        lst.append(pos)
counter = collections.Counter(lst)
for pos, count in counter.most_common(3):
    print(f"{pos}: {count}건")

명사: 60건
(미상): 19건
어미: 1건


실행결과

김치:328건
라면:86건
만두:89건
김밥:39건
국수:227건
떡볶이:24건
불고기:38건
비빔밥:38건




명사: 60건
(미상): 19건
어미: 1건

코드설명

1. 리스트를 순회하면서 검색어의 전체 결과 수를 프린트한다.

2.빈 리스트에 pos를 리스트를 순회하면서 append한다. 이때 pos는 .get을 통해 품사를 반환받는다. 혹은 없을 경우 미상을 받는다. collections.Counter를 이용하여 리스트 형식으로 각 품사의 횟수를 세고 counter.mostcommon(3) 을 이용하여 가장 많이 등장한 품사 세개를 센다.

가장 흔한 품사는 명사이며 이는 위의 단어들의 가장 대표되는 의미가 명사들이기 때문이다.